HOWTO.md

```diff
@@ -86,6 +86,66 @@ This will result into this module header:
                                                                    *your_module*
 ```
 
+## Config
+
+You can configure docgen on file basis. For example you can define how `functions` or `classes`
+are sorted.
+
+```lua
+---@config { ['function_order'] = 'ascending', ['class_order'] = 'descending' }
+```
+
+Available keys value pairs are:
+- `function_order`:
+  - `file_order` (default)
+  - `ascending`
+  - `descending`
+  - If you have a typo it will do `file_order` sorting
+- `class_order`:
+  - `file_order` (default)
+  - `ascending`
+  - `descending`
+  - If you have a typo it will do `file_order` sorting
+<!----> TODO(conni2461): DO WE WANT THIS? IF YES IMPLEMENT IT
+<!-- - `field_order`: -->
+<!--   - `file_order` (default) -->
+<!--   - `ascending` -->
+<!--   - `descending` -->
+<!--   - If you have a typo it will do `file_order` sorting -->
+
+## Class
+
+You can define your own classes and types to give a better sense of the Input of another function.
+Another good usecase for this are structs defined by ffi.
+
+Input:
+
+```lua
+---@class Array : Map @number indexed starting at 1
```

Comment:
I find this example kind of confusing. I don't understand exactly what it's going for.

Reply:
Let me write one from sql.nvim. Maybe i shouldn't have just copied the test

Reply:
I think i have found a pretty elaborate example which should make the use case pretty clear :)